### ISSUE TITLE
No Healthcheck in KV store (and Web) provider is configured

### ISSUE THREAD

**Issue by tmonteyne** [containous/traefik#1967]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Healthcheck is not enable when we used provider KV (zookeeper or consul) and provider web

### What did you expect to see?

No healthcheck 
The key / value pair is present 

ex for provider consul or zookeeper : `traefik_configurations/1/backends/backend1/healthcheck/path`
ex for provider web : `"healthcheck": {"path": "/health"}`

### What did you see instead?

Just healthcheck in `traefik.log` and backend eviction when the healthcehck is down 
Like : provider file 

log ex  with file config (rules.toml) . It's work fine
```shell
time="2017-08-11T15:10:42+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 647.079µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 684.532µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 5.484609ms"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 708.717µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 671.207µs"
time="2017-08-11T15:10:49+02:00" level=debug msg="Refreshing healthcheck for currentBackend backend1 "
time="2017-08-11T15:10:49+02:00" level=warning msg="HealthCheck is still failing [http://192.168.0.39:8082]"
```
### Output of `traefik version`: (_What version of Traefik are you using?_)

1.3.5

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Just activate KV store (or web) configuration

### If applicable, please paste the log output in debug mode (`--debug` switch)

log ex if i stop one of the three server when kv store (or web)  is configured  :
```shell
time="2017-08-11T15:06:53+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 1.124324ms"
time="2017-08-11T15:06:54+02:00" level=warning msg="Error forwarding to http://192.168.0.39:8082, err: dial tcp 192.168.0.39:8082: getsockopt: connection refused"
time="2017-08-11T15:06:55+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 754.716µs"
time="2017-08-11T15:06:55+02:00" level=warning msg="Error forwarding to http://192.168.0.39:8082, err: dial tcp 192.168.0.39:8082: getsockopt: connection refused"
time="2017-08-11T15:06:55+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 723.493µs"
time="2017-08-11T15:06:56+02:00" level=warning msg="Error forwarding to http://192.168.0.39:8082, err: dial tcp 192.168.0.39:8082: getsockopt: connection refused"
time="2017-08-11T15:06:56+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 916.527µs" (edited)
```
3:11
log ex  with file config (rules.toml) . It's work fine
```shell
time="2017-08-11T15:10:42+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 647.079µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 684.532µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 5.484609ms"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8081, code: 200, duration: 708.717µs"
time="2017-08-11T15:10:43+02:00" level=debug msg="Round trip: http://192.168.0.39:8083, code: 200, duration: 671.207µs"
time="2017-08-11T15:10:49+02:00" level=debug msg="Refreshing healthcheck for currentBackend backend1 "
time="2017-08-11T15:10:49+02:00" level=warning msg="HealthCheck is still failing [http://192.168.0.39:8082]"
```
3:14
The healthcheck runs in file mode (`rules.toml`) but not in key/value store mode with consul

**Comment by tbouvet** [comment]
Same problem with **v1.4.0-rc3.**

In consul environment the key **port** is not supported by traefik.

Example with this traefik configuration in consul K/V store:
```
"traefik/backends/backend1/healthcheck/path": "/ping"
"traefik/backends/backend1/healthcheck/interval": "10s"
"traefik/backends/backend1/healthcheck/port": 8080
```

The traefik log is:
```
time="2017-09-19T13:09:53Z" level=debug msg="Setting up backend health check [Path: /ping Port: 0 Interval: 10s]"
```

The Port variable is not OK.

**Comment by softkot** [comment]
The issue is still exists in 1.5-rc5. I had waste a ton of time until have understood the difference behavoir brtween file and KV (etcd) backends an finally faced to this issue report.

**Comment by ldez** [comment]
Its fixed #2652
